numpy/lib/utils.py

```diff
@@ -105,6 +105,20 @@ def newfunc(*args,**kwds):
         if doc is None:
             doc = depdoc
         else:
+            lines = doc.expandtabs().split('\n')
+            indent = _get_indent(lines[1:])
+            if lines[0].lstrip():
+                # Indent the original first line to let inspect.cleandoc()
+                # dedent the docstring despite the deprecation notice.
+                doc = indent * ' ' + doc
+            else:
+                # Remove the same leading blank lines as cleandoc() would.
+                skip = len(lines[0]) + 1
+                for line in lines[1:]:
+                    if len(line) > indent:
```

Comment:
Should this be looking at `lstrip`ed lines?

Is there any reason the first line is treated specially here?

Reply:
Edit: nevermind, this matches `cleandoc`